src/sgame/sg_bot_nav.cpp

```diff
@@ -586,9 +586,24 @@ static bool BotShouldJump( gentity_t *self, const gentity_t *blocker, const glm:
 	//check if jumping will clear us of entity
 	trap_Trace( &trace, origin, playerMins, playerMaxs, end, self->s.number, MASK_SHOT, 0 );
 
+	if ( blocker->s.modelindex == BA_A_BARRICADE )
+	{
+		glm::vec3 mins, maxs;
+		BG_BoundingBox( static_cast<buildable_t>( blocker->s.modelindex ), &mins, &maxs );
+		if ( G_Team( self ) == TEAM_ALIENS )
+		{
+			maxs.z *= BARRICADE_SHRINKPROP;
+		}
+		glm::vec3 pos = VEC2GLM( blocker->s.origin );
+		BG_BoundingBox( pClass, &playerMins, &playerMaxs, nullptr, nullptr, nullptr );
+		if ( origin.z + playerMins.z >= pos.z + maxs.z )
+		{
+			return true;
+		}
+	}
+
 	//if we can jump over it, then jump
-	//note that we also test for a blocking barricade because barricades will collapse to let us through
-	return blocker->s.modelindex == BA_A_BARRICADE || trace.fraction == 1.0f;
+	return trace.fraction == 1.0f;
```

Comment:
What if you just changed this to `return (blocker->s.modelindex == BA_A_BARRICADE && G_Team(self) == TEAM_ALIENS ) || trace.fraction == 1.0f;` ?

Reply:
You mean keeping the old code, that breaks in many situations, like when barricade is placed so high that even shrunk, a tyrant can't pass it?